Tests/TuistCacheTests/Cache/CacheRemoteStorageTests.swift

```diff
@@ -0,0 +1,161 @@
+    }
+
+    // - fetch
+
+    func test_fetch_whenClientReturnsAnError() throws {
+        // Given
+        let cloudClient = MockCloudClienting<CloudResponse<CloudCacheResponse>>()
+        cloudClient.configureForError(error: CloudClientError.unauthorized)
+        subject = CacheRemoteStorage(cloudClient: cloudClient)
+
+        // When
+        let result = subject.fetch(hash: "acho tio", userConfig: .test())
+            .toBlocking()
+            .materialize()
+
+        // Then
+        switch result {
+        case .completed:
+            XCTFail("Expected result to complete with error, but result was successful.")
+        case let .failed(_, error):
+            XCTAssertEqual(error as! CloudClientError, CloudClientError.unauthorized)
+        }
+    }
+
+    func test_fetch_whenClientReturnsASuccess() throws {
+        // Given
+        let httpResponse: HTTPURLResponse = .test()
+        let cacheResponse = CloudCacheResponse(url: .test(), expiresAt: 123)
+        let cloudResponse = CloudResponse<CloudCacheResponse>(status: "shaki", data: cacheResponse)
+        cloudClient = MockCloudClienting.makeForSuccess(object: cloudResponse, response: httpResponse)
+        subject = CacheRemoteStorage(cloudClient: cloudClient)
+
+        // When
+        let result = try subject.fetch(hash: "acho tio", userConfig: .test())
+            .toBlocking()
+            .single()
+
+        // Then
+        XCTAssertEqual(result, AbsolutePath("/"))
+    }
+
+    // - store
+
+    func test_store_whenClientReturnsAnError() throws {
+        // Given
+        let cloudClient = MockCloudClienting<CloudResponse<CloudCacheResponse>>()
+        cloudClient.configureForError(error: CloudClientError.unauthorized)
+        subject = CacheRemoteStorage(cloudClient: cloudClient)
+
+        // When
+        let result = subject.store(hash: "acho tio", userConfig: .test(), xcframeworkPath: .root)
+            .toBlocking()
+            .materialize()
+
+        // Then
+        switch result {
+        case .completed:
+            XCTFail("Expected result to complete with error, but result was successful.")
+        case let .failed(_, error):
+            XCTAssertEqual(error as! CloudClientError, CloudClientError.unauthorized)
+        }
+    }
+
+    func test_store_whenClientReturnsASuccess() throws {
+        // Given
+        let httpResponse: HTTPURLResponse = .test()
+        let cacheResponse = CloudCacheResponse(url: .test(), expiresAt: 123)
+        let cloudResponse = CloudResponse<CloudCacheResponse>(status: "shaki", data: cacheResponse)
+        cloudClient = MockCloudClienting.makeForSuccess(object: cloudResponse, response: httpResponse)
+        subject = CacheRemoteStorage(cloudClient: cloudClient)
+
+        // When
+        _ = try subject.store(hash: "acho tio", userConfig: .test(), xcframeworkPath: .root)
+            .toBlocking()
+            .last()
+
+        // Then
+//        XCTAssertEqual(result, [AbsolutePath("/")])
```

Comment:
???

Reply:
Yeah don't worry that's still WIP, it will not be merged like this. It's there so I can get potential feedback/tips on how to test functions that return RxSwift objects.